src/services/http/patterns.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcryptjs')
+const { crypt } = require('../../config/vars')
+const { responseHandler, errorHandler } = require('../../services/http/utils/response')
+
+/**
+ * Plugin handles the user entity data store
+ */
+module.exports = function plugin (options) {
+  var seneca = this
+
+  /**
+     * Handles creating a new administrator
+     */
+  seneca.add('role:auth,cmd:signup', (msg, done) => {
+    let args = msg.args.body
+    try {
+      bcrypt.genSalt(crypt.SALT_WORK_FACTOR, function (err, salt) {
+        bcrypt.hash(args.password, salt, function (err, hash) {
+          seneca.act('role:storage,cmd:insert', ({ type: 'administrator', input: { username: args.username, password: hash } }), function (err, res) {
+            console.log(res)
+            delete res.password
+            done(err, responseHandler(res))
+          })
+        })
+      })
+    } catch (err) {
```

Comment:
do you think the try/catch can capture the errors when calling `bcrypt.genSalt` and `bcrypt.hash`?

Reply:
You’re right , would be better to async or isolate the process to ensure the right error is being called up the chain